### PR TITLE
feat/form builder json download

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

Adds a download file button

# Test instructions | Instructions pour tester la modification

- Visit the `JSON` page / tab --- press download 
- This should download a file containing the form JSON

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [ ] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: components/form-builder/layout/DownloadFileButton.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  date = new Date(date.getTime() - offset * 60 * 1000);
+  return date.toISOString().split("T")[0];
+};
+
+export const DownloadFileButton = () => {
+  const { getSchema, form } = useTemplateStore();
+  const dowloadFile = useCallback(async () => {
+    async function retrieveFileBlob() {
+      try {
+        const blob = new Blob([getSchema()], { type: "text/plain" });
```
Comment (pcraig3) [inline components/form-builder/layout/DownloadFileButton.tsx:24]: Any reason not to make this `"application/json"`?

**Review by pcraig3 — APPROVED** [review]
Just a couple comments, otherwise this works well.

File: components/form-builder/layout/DownloadFileButton.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        const url = URL.createObjectURL(blob);
+        const a = document.createElement("a");
+        a.href = url;
+        a.download = slugify(`${form.titleEn}-${getDate()}`);
```
[the remark this answers is not in the document]
Comment (timarney) [reply components/form-builder/layout/DownloadFileButton.tsx:28]: ya thinking about adding time as well --- was thinking the shorter filename is possibly easier to spot --- also yes to the 1,2,3 thinking the OS handling is fine.

**Review by pcraig3 — APPROVED** [review]
Works perfect 👍

**Pull request merged**